watchers/configmap_watcher.go

```diff
@@ -0,0 +1,78 @@
+package watcher
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func WatchForChanges(clientset *kubernetes.Clientset, isLocalRun bool, namespace string, mgrConfigMapName string, dataField string, dataValue string, mutex *sync.Mutex) {
+	logger := log.FromContext(context.TODO())
+	logger.V(0).Info("watchForChanges", "namespace", namespace, "data field", dataField, "current value", dataValue)
+	logger.V(1).Info("watchForChanges", "namespace", namespace, "data field", dataField, "current value", dataValue)
+	for {
+		watcher, err := clientset.CoreV1().ConfigMaps(namespace).Watch(context.TODO(),
```

Comment:
Is `Watch` a blocking call ?

Reply:
IIUC, `Watch` is a blocking call. It waits for events on the `Event` channel. 
Is there anything you're not convinced of?

Reply:
this call is running in an infinite loop (line 20). When the watcher is initialized, it returns the `watcher.ResultChan()`.
Next, there is another infinite for loop (line 32) that reads from that channel. 

My question is why the outer infinite loop is required ? In which case would we need to re-initialize the watcher ?

Reply:
The `Watch` function is implemented by a `Get` call to the Kubernetes API server. So the `Watch` is a blocking call but the API server may close the close due to a timeout. So the infinite loop in line 20 handles this case. Is that reasonable for you?

Reply:
yes. thanks for the explanation.